src/app/children/children.service.ts

```diff
@@ -198,4 +274,23 @@ export class ChildrenService {
     );
   }
 
+  getCurrentSchool(childId: string): Promise<School> {
+    return this.queryLatestRelation(childId)
+      .then(relation => {
+        if (relation) {
+         return this.entityMapper.load<School>(School, relation.schoolId)
+        }
+        return null;
+      });
+  }
+
+  async getViewableSchools(childId: string): Promise<SchoolWithRelation[]> {
```

Comment:
In which sense are the Schools returned by `getViewableSchools()` "viewable"? Maybe we could name this in a more obvious way? The code looks like it loads all schools of a child?

Reply:
Maybe you could call it just getSchoolWithRelation. It's viewable in that sense, that you can display the name of the related school instead of just the schoolId which is saved in the relation.